### PR TITLE
NEXUS-21813 Helm can not add nexus hosted repository

### PR DESCRIPTION
**Pull request by mlukaretkyi**
https://issues.sonatype.org/browse/NEXUS-21813

### CODE REVIEW

File: nexus-repository-helm/src/main/java/org/sonatype/repository/helm/internal/createindex/CreateIndexFacetImpl.java [hunk 1]
```diff
@@ -112,9 +116,15 @@ public void on(RepositoryCreatedEvent createdEvent) {
   }
 
   private void maybeInvalidateIndex(final AssetEvent event) {
-     if(matchesRepository(event) && isEventRelevant(event)) {
-       invalidateIndex();
-     }
+    Asset asset = event.getAsset();
+    String formatName = asset.format();
+    if (formatName.equals(HelmFormat.NAME)) {
```
Comment (proteus1121) [inline nexus-repository-helm/src/main/java/org/sonatype/repository/helm/internal/createindex/CreateIndexFacetImpl.java]: maybe better? HelmFormat.NAME.equals(formatName)

Comment (mlukaretkyi) [reply nexus-repository-helm/src/main/java/org/sonatype/repository/helm/internal/createindex/CreateIndexFacetImpl.java]: Yes, you are absolutely right! Fixed

File: nexus-repository-helm/src/main/java/org/sonatype/repository/helm/internal/HelmFacetImpl.java [hunk 2]
```diff
@@ -121,17 +122,21 @@ private Component findOrCreateComponent(final StorageTx tx,
...
   {
     Builder builder = builder()
         .where(P_COMPONENT).isNotNull();
+    if (assetKind != null) {
+      builder.and(P_ATTRIBUTES + "." + HelmFormat.NAME + "." + P_ASSET_KIND).eq(assetKind.name());
+    }
 
-    Query query = builder.build();
+    Query query = builder
+        .build();
```
Comment (chernikovmaksym) [inline nexus-repository-helm/src/main/java/org/sonatype/repository/helm/internal/HelmFacetImpl.java]: Formatting

File: nexus-repository-helm/src/test/java/org/sonatype/repository/helm/internal/createindex/CreateIndexServiceImplTest.java [hunk 3]
```diff
@@ -111,7 +112,7 @@ public void testBuildIndexYaml() throws Exception {
     shaMap.put("sha256", "12345");
 
     when(assetAttributes.get("checksum", Map.class)).thenReturn(shaMap);
-    when(helmFacet.browseComponentAssets(storageTx)).thenReturn(list);
+    when(helmFacet.browseComponentAssets(storageTx, AssetKind.HELM_PACKAGE)).thenReturn(list);
```
Comment (chernikovmaksym) [inline nexus-repository-helm/src/test/java/org/sonatype/repository/helm/internal/createindex/CreateIndexServiceImplTest.java]: static import

File: nexus-repository-helm/src/test/java/org/sonatype/repository/helm/internal/createindex/CreateIndexServiceImplTest.java [hunk 4]
```diff
@@ -124,7 +125,7 @@ public void testIndexYamlBuiltEvenWhenNoAssets() throws Exception {
     when(assets.iterator()).thenReturn(assetIterator);
     when(assetIterator.next()).thenReturn(asset);
     when(asset.componentId()).thenReturn(null);
-    when(helmFacet.browseComponentAssets(storageTx)).thenReturn(assets);
+    when(helmFacet.browseComponentAssets(storageTx, AssetKind.HELM_PACKAGE)).thenReturn(assets);
```
Comment (chernikovmaksym) [inline nexus-repository-helm/src/test/java/org/sonatype/repository/helm/internal/createindex/CreateIndexServiceImplTest.java]: static import

File: nexus-repository-helm/src/main/java/org/sonatype/repository/helm/internal/createindex/CreateIndexServiceImpl.java [hunk 5]
```diff
@@ -75,7 +80,7 @@ public TempBlob buildIndexYaml(final Repository repository) {
 
     ChartIndex index = new ChartIndex();
 
-    for (Asset asset : helmFacet.browseComponentAssets(tx)) {
+    for (Asset asset : helmFacet.browseComponentAssets(tx, AssetKind.HELM_PACKAGE)) {
```
Comment (chernikovmaksym) [inline nexus-repository-helm/src/main/java/org/sonatype/repository/helm/internal/createindex/CreateIndexServiceImpl.java]: static import

File: nexus-repository-helm/src/main/java/org/sonatype/repository/helm/internal/createindex/CreateIndexServiceImpl.java [hunk 6]
```diff
@@ -93,8 +98,23 @@ private void parseAssetIntoChartEntry(final ChartIndex index, final Asset asset)
...
+          .map(param -> {
+            String[] split = param.split(HelmAttributes.MAINTAINERS_SEPARATOR);
+            String key = split[0];
+            String value = split[1];
+            return Collections.singletonMap(key, value);
+          })
+          .collect(Collectors.toList());
+      chartEntry.setMaintainers(nextMaintainers);
+    }
+
```
Comment (sergeyonofreychuck) [inline nexus-repository-helm/src/main/java/org/sonatype/repository/helm/internal/createindex/CreateIndexServiceImpl.java]: move this block to a separate function

Comment (mlukaretkyi) [reply nexus-repository-helm/src/main/java/org/sonatype/repository/helm/internal/createindex/CreateIndexServiceImpl.java]: Fixed. 
https://github.com/sonatype-nexus-community/nexus-repository-helm/pull/92/commits/fcb1480ed325222a09c2cd0124cd92f67e7bf676

**Pull request merged**